### PR TITLE
feat: account presence based on aex9 balance

### LINKED ISSUE #186: AEX-9 balances for account don't seem consistent
https://testnet.aeternity.io/mdw/aex9/balances/ct_BxckqeNbTYH7ayohwsnyq8JeU6G49zDrZuLoVXjE4fKwzcyna

shows balance for account `ak_11111111111111111111111111111111273Yts`

https://testnet.aeternity.io/mdw/aex9/balances/account/ak_11111111111111111111111111111111273Yts

doesn't show any balances

### PR DESCRIPTION
**Pull request by jyeshe**
## What

Once transfer event is something manually emitted from the contract code, account presence might be updated after any transfer as a best effort to keep track of 'account-to-contract' mapping.

## Why

Fixes #186 as a best effort as the complete mapping (of all accounts with balance in the contract) depends on contract code to emit the event.

An account might appear in:
/aex9/balances/:contract_id
but not in:
/aex9/balances/account/:account_id

## Validation steps

1. With `master` branch, this account is initially not mapped to any contract (before running the migration, which runs automatically):
http://localhost:4000/aex9/balances/account/ak_2CuCtmoExsuW7iG5BbRAoncSZAoZt8Abu21qrPUhqfFNin75iL
http://localhost:4000/aex9/balances/account/ak_eYHyfKSZiU3nzDTJBkrXAzVszoxDU4DHDw5xz6f2i5YVME234
2. Run the Mdw with `186-index-account-presence` branch. Wait for migration to get all aex9 contracts balances and update the accounts presence (this might take about 10 minutes).
3. Repeat the requests of step 1 and the mapped accounts balance is shown:
image
image

Expected migration indexed count is (for Mdw snapshot of August 24th 2021):
Indexed 14 records in 504s

## Additional Notes

Please notice the actual transfer calls are not shown in the request as there is no txi (mapping not after a transfer event with the account).

These aex9 contracts are out of the scope of this PR as a call to get their store/balance returns "Out of gas" error:

Out of gas for ct_2M4mVQCDVxu6mvUrEue1xMafLsoA1bgsfC3uT95F3r1xysaCvE
Out of gas for ct_21oSi6uMMcnnyxc1KTP8MReBPVvBQQ9oeoA7aexQBsF2o16HwT
Out of gas for ct_2czjMpbeeds9Geb272UfiEvhQHc8QZM5AM8jngwpqdMnWNDP3h
Out of gas for ct_25JPViJPDPjroiL1wr1ngWJ8pK5ykN7x3LqGMWPCuC6PtLjzKT
Out of gas for ct_2mzKUzT55wz4wYUmH4s3dHf3YBprDLi92uxvEah8fe3BaLfkBU
Out of gas for ct_2bFV4kxtmUeKTF5Eb5STfnAx6UgxzBqTL1pYLF1GBhcmoQAhLf
Out of gas for ct_BmCPDh6TK3KtWdwKEJrSCnrvghbkjekuTTh7B5tLGduF6rTFG
Out of gas for ct_227nuD7YiHL7pJ63FP1CnbhKgU9ueWKP77sA26huE47s31tuqQ
Out of gas for ct_jyw2fRpr6srNwSqE6FJFyZSD1upD91eZ5ZZKZfBhuDmBxTU53
Out of gas for ct_VxFemyM2ML15ApWrFqBKrarVbNfBASPBs1SC1zMuwaWrnrTSy
Out of gas for ct_2GPhKGbRxZaaJtqRTnxLk5UDSNNEtRt3tyWuKVQeoVrerRfhHa
Out of gas for ct_7UfopTwsRuLGFEcsScbYgQ6YnySXuyMxQWhw6fjycnzS5Nyzq
Out of gas for ct_M9yohHgcLjhpp1Z8SaA1UTmRMQzR4FWjJHajGga8KBoZTEPwC
Out of gas for ct_BwJcRRa7jTAvkpzc2D16tJzHMGCJurtJMUBtyyfGi2QjPuMVv
Out of gas for ct_uGk1rkSdccPKXLzS259vdrJGTWAY9sfgVYspv6QYomxvWZWBM
Out of gas for ct_26GAggwrHJLw45vopxu1HA51t6pw4xjzqvq66kh1WPtZjNXMyL

### CODE REVIEW

File: lib/ae_mdw_web/controllers/aex9_controller.ex [hunk 1]
```diff
@@ -300,47 +305,6 @@ defmodule AeMdwWeb.Aex9Controller do
 
   defp top?(conn), do: presence?(conn, "top")
 
-  defp normalize_balances(bals) do
```
Comment (jyeshe) [inline lib/ae_mdw_web/controllers/aex9_controller.ex:303]: moved to lib/ae_mdw_web/helpers/aex9_helper.ex

File: lib/ae_mdw_web/controllers/aex9_controller.ex [hunk 2]
```diff
@@ -300,47 +305,6 @@ defmodule AeMdwWeb.Aex9Controller do
...
-    for {{:address, pk}, amt} <- bals, reduce: %{} do
-      acc ->
-        Map.put(acc, :aeser_api_encoder.encode(:account_pubkey, pk), amt)
-    end
-  end
-
-  defp balance_to_map({amount, txi, contract_pk}) do
```
Comment (jyeshe) [inline lib/ae_mdw_web/controllers/aex9_controller.ex:310]: moved to lib/ae_mdw_web/views/aex9_view.ex

File: lib/ae_mdw_web/controllers/aex9_controller.ex [hunk 3]
```diff
@@ -363,32 +327,9 @@ defmodule AeMdwWeb.Aex9Controller do
     )
   end
 
-  defp transfer_to_map({recipient_pk, sender_pk, amount, call_txi, log_idx}, :rev_aex9_transfer),
```
Comment (jyeshe) [inline lib/ae_mdw_web/controllers/aex9_controller.ex:366]: moved to lib/ae_mdw_web/views/aex9_view.ex

**Review by sborrazas — CHANGES_REQUESTED** [review]
Looks good, just some minor adjustments perhaps?

**Comment by thepiwo** [comment]
Do we have some idea on the performance hit of these changes?

**Comment by jyeshe** [comment]
> Do we have some idea on the performance hit of these changes?

It's quite slow. The sync process logic works sequentially but well reminded, this specific part we might probably run in parallel. Thanks

**Review by sborrazas — APPROVED** [review]

**Review by thepiwo — APPROVED** [review]

**Pull request merged**